adapters/resetdigital/resetdigital.go

```diff
@@ -4,305 +4,237 @@ import (
-		rdImp.MediaTypes.Audio.Mimes = append(rdImp.MediaTypes.Audio.Mimes, imp.Audio.MIMEs...)
+
+	uri := a.endpoint
+	if resetDigitalExt.PlacementID != "" {
+		uri = fmt.Sprintf("%s?pid=%s", a.endpoint, resetDigitalExt.PlacementID)
 	}
 
-	var bidderExt adapters.ExtImpBidder
-	var resetDigitalExt openrtb_ext.ImpExtResetDigital
+	reqHeaders := baseHeaders.Clone()
 
-	if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
-		return resetDigitalRequest{}, err
-	}
-	if err := json.Unmarshal(bidderExt.Bidder, &resetDigitalExt); err != nil {
-		return resetDigitalRequest{}, err
+	reqs := []*adapters.RequestData{
+		{
+			Method:  http.MethodPost,
+			Uri:     uri,
+			Body:    reqBody,
+			Headers: reqHeaders,
+			ImpIDs:  []string{imp.ID},
+		},
 	}
-	rdImp.ZoneID.PlacementID = resetDigitalExt.PlacementID
-
-	reqData.Imps = append(reqData.Imps, rdImp)
 
-	return reqData, nil
+	return reqs, errs
 }
 
 func (a *adapter) MakeBids(request *openrtb2.BidRequest, requestData *adapters.RequestData, responseData *adapters.ResponseData) (*adapters.BidderResponse, []error) {
-	if adapters.IsResponseStatusCodeNoContent(responseData) {
+	if responseData.StatusCode == http.StatusNoContent {
 		return nil, nil
 	}
 
-	if err := adapters.CheckResponseStatusCodeForErrors(responseData); err != nil {
-		return nil, []error{err}
+	if responseData.StatusCode >= http.StatusBadRequest && responseData.StatusCode < http.StatusInternalServerError {
+		return nil, []error{&errortypes.BadInput{
+			Message: fmt.Sprintf("Unexpected status code: %d. Run with request.debug = 1 for more info", responseData.StatusCode),
+		}}
 	}
 
-	var response resetDigitalBidResponse
-	if err := json.Unmarshal(responseData.Body, &response); err != nil {
-		return nil, []error{err}
+	if responseData.StatusCode != http.StatusOK {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Unexpected status code: %d. Run with request.debug = 1 for more info", responseData.StatusCode),
+		}}
 	}
 
-	if len(response.Bids) != 1 {
-		return nil, []error{fmt.Errorf("expected exactly one bid in the response, but got %d", len(response.Bids))}
+	var bidResp openrtb2.BidResponse
+	if err := json.Unmarshal(responseData.Body, &bidResp); err != nil {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Bad server response: %s", err),
+		}}
 	}
 
-	resetDigitalBid := &response.Bids[0]
-
-	requestImp, found := findRequestImpByID(request.Imp, resetDigitalBid.ImpID)
-	if !found {
-		return nil, []error{fmt.Errorf("no matching impression found for ImpID %s", resetDigitalBid.ImpID)}
+	if len(bidResp.SeatBid) == 0 {
+		return nil, nil
 	}
 
-	bid, err := getBidFromResponse(resetDigitalBid)
-	if err != nil {
-		return nil, []error{err}
+	return parseBidResponse(request, &bidResp)
+}
+
+func parseBidResponse(request *openrtb2.BidRequest, bidResp *openrtb2.BidResponse) (*adapters.BidderResponse, []error) {
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(5)
```

Comment:
Nitpick: the `len(bidResp.SeatBid) == 0` check in line 151 above, makes an initialization with the exact lenght of `bidResp.SeatBid` safe here.
```
158   func parseBidResponse(request *openrtb2.BidRequest, bidResp *openrtb2.BidResponse) (*adapters.BidderResponse, []error) {
159 -     bidResponse := adapters.NewBidderResponseWithBidsCapacity(5)
    +     bidResponse := adapters.NewBidderResponseWithBidsCapacity(len(bidResp.SeatBid))
160       var errs []error
161  
162       if bidResp.Cur != "" {
adapters/resetdigital/resetdigital.go
```